Document same-role permissions for team-based access control (#728)
mautic/user-documentation: docs/users_roles/managing_roles.rst

  
  For example, if a User's Role doesn't have Asset permissions, they can't create or view widgets on the dashboard for Asset data.

You can also create :xref:`Roles using the API`.

.. vale off

> Using same-role permissions

Author: Added new "Using same-role permissions" section based on [PR #15465](https://github.com/mautic/mautic/pull/15465), which introduces `viewsamerole`, `editsamerole`, `deletesamerole`, and `publishsamerole` permission levels across Assets, Campaigns, Channels, Dynamic Content, Emails, Forms, Landing Pages, Reports, and Segments. The Italy/Germany team example comes directly from the PR description.

Source: https://github.com/mautic/mautic/pull/15465

> Using same-role permissions

Reviewer: ```suggestion
Using Same Role permissions
```

===========================

.. vale on

> Same-role permissions provide an intermediate access level between **Own** and **Others**. This allows team-based collaboration where Users can share resources within their team without granting access to resources from other teams.

Reviewer: ```suggestion
Same Role permissions provide an intermediate access level between **Own** and **Others**. This allows team-based collaboration where Users can share resources within their team without granting access to resources from other teams.
```

When same-role permissions apply

> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
.. vale off

When Same Role permissions apply
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. vale on
```

> Same-role permissions apply to the following areas of Mautic:

Reviewer: ```suggestion
Same Role permissions apply to the following areas of Mautic:
```

* Assets
* Campaigns
* Channels - Marketing Messages
* Dynamic Content
* Emails
* Focus Items
* Forms
* Landing Pages
* Reports
* Segments

For each area, you can set **View Same Role**, **Edit Same Role**, **Delete Same Role**, and **Publish Same Role** permissions.

Example: regional teams sharing content
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> A company has marketing teams in multiple countries. Each team needs to collaborate on shared content, but shouldn't access other teams' work.

Reviewer: Company here is general and not Mautic feature. Wrap in vale off/on statements.

```suggestion
.. vale off

A company has marketing teams in multiple countries. Each team needs to collaborate on shared content, but shouldn't access other teams' work.

.. vale on
```

1. Create a Role called ``Italy`` and assign all Italian team members to it.

2. Create a Role called ``Germany`` and assign all German team members to it.

> 3. For each Role, enable the same-role permissions. For example, select **View Same Role**, **Edit Same Role**, and **Publish Same Role** for Emails and Campaigns.

Reviewer: Use # instead of numbers to follow Mautic style guide.

```suggestion
#. Create a Role called ``Italy`` and assign all Italian team members to it.
#. Create a Role called ``Germany`` and assign all German team members to it.
#. For each Role, enable the Same Role permissions. For example, select **View Same Role**, **Edit Same Role**, and **Publish Same Role** for Emails and Campaigns.
```

With this setup:

* Italian Users can view, edit, and publish Emails created by other Italian Users.
* Italian Users can't see or modify Emails created by German Users.
* German Users can view, edit, and publish Emails created by other German Users.
* German Users can't see or modify Emails created by Italian Users.

This pattern works for any scenario where groups need internal collaboration with isolation from other groups, such as regional offices, product lines, client accounts, or business units.

Combining permission levels
~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Same-role permissions work alongside Own and Others permissions. Mautic checks permissions in this order:

Reviewer: ```suggestion
Same Role permissions work alongside Own and Others permissions. Mautic checks permissions in this order:
```

1. If a User has **Others** permission, they can access all resources regardless of who created them.

2. If a User has **Same Role** permission, they can access resources created by themselves and by other Users sharing the same Role.

> 3. If a User has only **Own** permission, they can access only resources they created themselves.

Reviewer: ```suggestion
#. If a User has **Others** permission, they can access all resources regardless of who created them.
#. If a User has **Same Role** permission, they can access resources created by themselves and by other Users sharing the same Role.
#. If a User has only **Own** permission, they can access only resources they created themselves.
```

  
  For example, if a User's Role doesn't have Asset permissions, they can't create or view widgets on the dashboard for Asset data.

You can also create :xref:`Roles using the API`.

.. vale off

Using Same Role permissions

> ===========================

Reviewer: ```suggestion
Using 'Same Role' permissions
=============================
```

.. vale on

Same Role permissions provide an intermediate access level between **Own** and **Others**. This allows team-based collaboration where Users can share resources within their team without granting access to resources from other teams.

.. vale off

When Same Role permissions apply

> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: @promptless-for-oss, looking at the content, which one is suitable here? "Where" or "when"?

> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reviewer: ```suggestion
Where 'Same Role' permissions apply
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

.. vale on

Same Role permissions apply to the following areas of Mautic:

* Assets
* Campaigns
* Channels - Marketing Messages
* Dynamic Content
* Emails
* Focus Items
* Forms
* Landing Pages
* Reports
* Segments

> For each area, you can set **View Same Role**, **Edit Same Role**, **Delete Same Role**, and **Publish Same Role** permissions.

Reviewer: @andersonjeccel I believe we're now using 'Activate or Turn On' instead of 'Publish' as per Vale warning here? Although **Publish Same Role** is correct in terms of how it renders on the UI, Vale will always warn us about the correct term to use and this has been inconsistent throughout the docs.

I'm wondering, is there any plan to update the UI in the near future? Or how should we address this in the docs?

Reviewer: @escopecz do you have any thoughts on this one?

Reviewer 2: It seems we are still using the "publish" word in the permission for all the entities. I'm not sure how to deal with that. Perhaps update them to be the same as the button for each entity?

@andersonjeccel

Reviewer 3: In the roles it's still using Publish - this is something we would perhaps have to consider altering in the product to align with the changes that were made elsewhere. For the purposes of this PR, leave at Publish. One for @mautic/ux-ui-tiger-team to follow up with, if needed.

> For each area, you can set **View Same Role**, **Edit Same Role**, **Delete Same Role**, and **Publish Same Role** permissions.

Reviewer: We need to keep the UI term as is.

```suggestion
.. vale off

For each area, you can set **View Same Role**, **Edit Same Role**, **Delete Same Role**, and **Publish Same Role** permissions.

.. vale on
```

Example: regional teams sharing content
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. vale off

A company has marketing teams in multiple countries. Each team needs to collaborate on shared content, but shouldn't access other teams' work.

.. vale on

> #. Create a Role called ``Italy`` and assign all Italian team members to it.
> #. Create a Role called ``Germany`` and assign all German team members to it.
> #. For each Role, enable the Same Role permissions. For example, select **View Same Role**, **Edit Same Role**, and **Publish Same Role** for Emails and Campaigns.

Reviewer: ```suggestion
.. vale off

#. Create a Role called ``Italy`` and assign all Italian team members to it.
#. Create a Role called ``Germany`` and assign all German team members to it.
#. For each Role, enable the Same Role permissions. For example, select **View Same Role**, **Edit Same Role**, and **Publish Same Role** for Emails and Campaigns.

.. vale on
```

With this setup:

> * Italian Users can view, edit, and publish Emails created by other Italian Users.

Reviewer: ```suggestion
* Italian Users can view, edit, and send Emails created by other Italian Users.
```

* Italian Users can't see or modify Emails created by German Users.

> * German Users can view, edit, and publish Emails created by other German Users.

Reviewer: ```suggestion
* German Users can view, edit, and send Emails created by other German Users.
```

* German Users can't see or modify Emails created by Italian Users.

This pattern works for any scenario where groups need internal collaboration with isolation from other groups, such as regional offices, product lines, client accounts, or business units.

Combining permission levels
~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Same Role permissions work alongside Own and Others permissions. Mautic checks permissions in this order:

Reviewer: ```suggestion
Same Role permissions work alongside **Own** and **Others** permissions. Mautic checks permissions in this order:
```
